pkg/util/informermanager/federatedinformermanager_test.go

```diff
@@ -0,0 +1,1265 @@
+		g.Expect(exists).To(gomega.BeFalse())
+		g.Expect(client).To(gomega.BeNil())
+	})
+
+	t.Run("clients for new clusters should be available eventually", func(t *testing.T) {
+		t.Parallel()
+
+		g := gomega.NewWithT(t)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// 1. Bootstrap environment
+
+		defaultFTCs := []*fedcorev1a1.FederatedTypeConfig{}
+		defaultObjs := map[string][]*unstructured.Unstructured{}
+		defaultClusters := []*fedcorev1a1.FederatedCluster{}
+		generators := []*EventHandlerGenerator{}
+		clusterHandlers := []*ClusterEventHandler{}
+		manager, fedClient := bootstrapFederatedInformerManagerWithFakeClients(g, ctx, defaultFTCs, defaultObjs, defaultClusters, generators, clusterHandlers)
+
+		// 2. Verify that client for cluster-1 does is not available initially.
+
+		g.Consistently(func(g gomega.Gomega) {
+			client, exists := manager.GetClusterClient("cluster-1")
+			g.Expect(exists).To(gomega.BeFalse())
+			g.Expect(client).To(gomega.BeNil())
+		}).WithTimeout(time.Second * 2).Should(gomega.Succeed())
+
+		// 3. Create a new cluster
+
+		cluster, err := fedClient.CoreV1alpha1().FederatedClusters().Create(ctx, getTestCluster("cluster-1"), metav1.CreateOptions{})
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+
+		// 4. Verify that client for new cluster is eventually available
+
+		g.Eventually(func(g gomega.Gomega) {
+			client, exists := manager.GetClusterClient(cluster.Name)
+			g.Expect(exists).To(gomega.BeTrue())
+			g.Expect(client).ToNot(gomega.BeNil())
+		}).WithTimeout(time.Second * 2).Should(gomega.Succeed())
+	})
+
+	t.Run("listers for existing FTCs and clusters should be available eventually", func(t *testing.T) {
+		t.Parallel()
+
+		g := gomega.NewWithT(t)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// 1. Bootstrap environment
+
+		defaultFTCs := []*fedcorev1a1.FederatedTypeConfig{deploymentFTC, configmapFTC, secretFTC}
+		defaultObjs := map[string][]*unstructured.Unstructured{}
+		defaultClusters := []*fedcorev1a1.FederatedCluster{
+			getTestCluster("cluster-1"),
+			getTestCluster("cluster-2"),
+			getTestCluster("cluster-3"),
+		}
+		generators := []*EventHandlerGenerator{}
+		clusterHandlers := []*ClusterEventHandler{}
+		manager, _ := bootstrapFederatedInformerManagerWithFakeClients(g, ctx, defaultFTCs, defaultObjs, defaultClusters, generators, clusterHandlers)
+
+		// 2. Verify that listers for existing FTCs and clusters are eventually available
+
+		for _, ftc := range defaultFTCs {
+			apiresource := ftc.GetSourceType()
+			gvr := schemautil.APIResourceToGVR(&apiresource)
+
+			for _, cluster := range defaultClusters {
+				g.Eventually(func(g gomega.Gomega) {
+					lister, informerSynced, exists := manager.GetResourceLister(gvr, cluster.Name)
+
+					g.Expect(exists).To(gomega.BeTrue())
+					g.Expect(lister).ToNot(gomega.BeNil())
+					g.Expect(informerSynced()).To(gomega.BeTrue())
+				})
```

Comment:
Missed assertion.

Reply:
Resolving this first since it is now captured by the linter